python3Packages.spacy: 2.0.16 -> 2.0.18

danieldk opened the pull request:
###### Motivation for this change

Update spaCy to 2.0.18. Also, it seems that we cannot build spaCy on Python 2 anymore, since one of its transitive dependencies (ftfy) does not work with Python 2 anymore.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

danieldk committed:
python3Package.thinc: 6.12.0 -> 6.12.1

danieldk committed:
python3Packages.spacy: 2.0.16 -> 2.0.18

Changes:

https://github.com/explosion/spaCy/releases/tag/v2.0.18
https://github.com/explosion/spaCy/releases/tag/v2.0.17

dotlambda commented on pkgs/development/python-modules/thinc/default.nix:
```diff
@@ -66,7 +66,8 @@ buildPythonPackage rec {
       --replace "pathlib==1.0.1" "pathlib>=1.0.0,<2.0.0" \
       --replace "plac>=0.9.6,<1.0.0" "plac>=0.9.6" \
-      --replace "wheel>=0.32.0,<0.33.0" "wheel>=0.31.0"
+      --replace "wheel>=0.32.0,<0.33.0" "wheel>=0.31.0" \
```
This line can be removed.

dotlambda reviewed (commented):
Please check if `propagatedBuildInputs` is still up to date: https://github.com/explosion/thinc/blob/v6.12.1/setup.py#L163-L177

danieldk commented:
I have moved some of the dependencies to `checkInputs` since they are only used for unit tests (which are disabled). I also removed two unused dependencies from the spacy derivation.
